### ISSUE TITLE
Twoway auth strange ui behavior

### ISSUE THREAD

**Issue by R4BNL** [keycloak/keycloak#20688]
### Before reporting an issue

- [X] I have searched existing issues
- [X] I have reproduced the issue with the latest release

### Area

authentication

### Describe the bug

In our SaaS solution we are using Keycloak 20.0.3. One of our customers is using Two way authentication but has four devices registered. There is a small issue where the selection button in UI is not correctly lined out making it very unclear for end users which device they are selecting. 

### Version

20.0.3

### Expected behavior

The button is shown correctly behind the device. 

### Actual behavior

The button is lined out in the top row. While the device is shown on the second row. 

### How to Reproduce?

1: Enable two way auth in Keycloak.
2: Link four devices to a users.  
3: See result when logging in.
![image](https://github.com/keycloak/keycloak/assets/135126238/4f11e2ef-5948-4836-85b5-b580ea71af97)

**Comment by rmartinc** [comment]
I didn't find a related issue either. But this is just a plain html/css issue. I would move the checkbox inside the box of the device.

For example on top:

```diff
diff --git a/themes/src/main/resources/theme/base/login/login-otp.ftl b/themes/src/main/resources/theme/base/login/login-otp.ftl
index a43778d900..29eca47d5d 100755
--- a/themes/src/main/resources/theme/base/login/login-otp.ftl
+++ b/themes/src/main/resources/theme/base/login/login-otp.ftl
@@ -9,8 +9,8 @@
                 <div class="${properties.kcFormGroupClass!}">
                     <div class="${properties.kcInputWrapperClass!}">
                         <#list otpLogin.userOtpCredentials as otpCredential>
-                            <input id="kc-otp-credential-${otpCredential?index}" class="${properties.kcLoginOTPListInputClass!}" type="radio" name="selectedCredentialId" value="${otpCredential.id}" <#if otpCredential.id == otpLogin.selectedCredentialId>checked="checked"</#if>>
                             <label for="kc-otp-credential-${otpCredential?index}" class="${properties.kcLoginOTPListClass!}" tabindex="${otpCredential?index}">
+                                <input id="kc-otp-credential-${otpCredential?index}" class="${properties.kcLoginOTPListInputClass!}" type="radio" name="selectedCredentialId" value="${otpCredential.id}" <#if otpCredential.id == otpLogin.selectedCredentialId>checked="checked"</#if>>
                                 <span class="${properties.kcLoginOTPListItemHeaderClass!}">
                                     <span class="${properties.kcLoginOTPListItemIconBodyClass!}">
                                       <i class="${properties.kcLoginOTPListItemIconClass!}" aria-hidden="true"></i>
@@ -55,4 +55,4 @@
             </div>
         </form>
     </#if>
```

![otp-above](https://github.com/keycloak/keycloak/assets/26901232/1c378980-f994-420c-9a36-a7390d63b206)

Or in the left (as it's now but inside the box):

```diff
diff --git a/themes/src/main/resources/theme/base/login/login-otp.ftl b/themes/src/main/resources/theme/base/login/login-otp.ftl
index a43778d900..5baf013c1c 100755
--- a/themes/src/main/resources/theme/base/login/login-otp.ftl
+++ b/themes/src/main/resources/theme/base/login/login-otp.ftl
@@ -9,9 +9,12 @@
                 <div class="${properties.kcFormGroupClass!}">
                     <div class="${properties.kcInputWrapperClass!}">
                         <#list otpLogin.userOtpCredentials as otpCredential>
-                            <input id="kc-otp-credential-${otpCredential?index}" class="${properties.kcLoginOTPListInputClass!}" type="radio" name="selectedCredentialId" value="${otpCredential.id}" <#if otpCredential.id == otpLogin.selectedCredentialId>checked="checked"</#if>>
                             <label for="kc-otp-credential-${otpCredential?index}" class="${properties.kcLoginOTPListClass!}" tabindex="${otpCredential?index}">
                                 <span class="${properties.kcLoginOTPListItemHeaderClass!}">
+                                    <span class="${properties.kcLoginOTPListItemTitleClass!}">
+                                      <input id="kc-otp-credential-${otpCredential?index}" class="${properties.kcLoginOTPListInputClass!}" type="radio" name="selectedCredentialId" value="${otpCredential.id}" <#if otpCredential.id == otpLogin.selectedCredentialId>checked="checked"</#if>>
+                                    </span>
+                                    <span class="${properties.kcLoginOTPListItemIconBodyClass!}"></span>
                                     <span class="${properties.kcLoginOTPListItemIconBodyClass!}">
                                       <i class="${properties.kcLoginOTPListItemIconClass!}" aria-hidden="true"></i>
                                     </span>
@@ -55,4 +58,4 @@
             </div>
         </form>
     </#if>
```

![otp-right](https://github.com/keycloak/keycloak/assets/26901232/ead6244d-e36e-4c14-a356-10495155ca8d)

@jonkoops @R4BNL anyone interested WDYT?

**Comment by avelarix** [comment]
Hello, i would prefer the first option.
The inside the box on top of the name.

**Comment by jonkoops** [comment]
I think that might be a valid fix yeah. I personally find the radio button on the same line as the text the best option, as this matches the default behavior of [`<input type="radio">`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/radio).

I'll defer UX decisions to our expert @xianli123 though.

**Comment by avelarix** [comment]
Hi,I just prefer the radio button on top because this way you can have 3 users on the same row and otherwise you can’t.And if i may i would say that the box height is too big for just one line of text in my opinion.Thank you both for taking care of this topic.If you have just one or two users ia ok, but if you have a team of 10 technicians that need access to this, it is not ok the big windows iObrigado José Avelar Sent from my iPhoneOn 15 Jun 2023, at 18:21, Jon Koops ***@***.***> wrote:﻿
I think that might be a valid fix yeah. I personally find the radio button on the same line as the text the best option, as this matches the default behavior of <input type="radio">.
I'll defer UX decisions to our expert @xianli123 though.

**Comment by xianli123** [comment]
Hi @jonkoops @avelarix @rmartinc @R4BNL Thanks for involving me. I am looking at the issue and looking for a solution.

**Comment by xianli123** [comment]
Hi @R4BNL I'm trying to replicate your initial share. But I failed. Can you share your steps for generating this interface? Thanks!

**Comment by rmartinc** [comment]
@xianli123 With a default keycloak, create or use an existing user, go to the account console (`/realms/<realm_name>/account/`), login with that user and in `Account Security` → `Signing in` click on `Setup an authenticator application`. You need FreeOTP or any OTP application that works with keycloak to scan the QR. Do the same 3 or 4 times to setup several devices like in the situation that @R4BNL describes.

Then logout from console, login again with the same user. After typing the password the OTP page is displayed and it shows the list of devices you have configured for OTP.

Regards!

**Comment by R4BNL** [comment]
@xianli123 There is not much to this, we use the default template of Keycloak 20. To reproduce this, you need to enable Two way authentication in keycloak. Then the user needs to have 4 or more devices added.

**Comment by xianli123** [comment]
@rmartinc @R4BNL Thanks. I've seen it. It looks a bit weird.

**Comment by xianli123** [comment]
Hi @jonkoops @avelarix @rmartinc @R4BNL, Thanks for your suggestions. You may know that the new console is followed the design system named **Patterfly 4**. According to Patterfly4's guideline, I used a radio button instead of using the card view. The screenshot below shows the UI. What do you think?

<img width="545" alt="Screenshot 2023-06-16 at 15 36 16" src="https://github.com/keycloak/keycloak/assets/53931519/a3dbddba-0a4c-456e-8833-7cbf32b42534">

**Comment by rmartinc** [comment]
@xianli123 If you want to simplify it and just use a radio, it's OK for me. I'm easy to convince. :smile:

Let's see what the others think. Probably this will need little tweaks in the CI tests.

**Comment by xianli123** [comment]
Thanks👍 @rmartinc

**Comment by jonkoops** [comment]
Yeah, I am all for keeping things simple where possible. I think the card view just makes things too crowded, the margins push in too much on the controls. So definitely in favor of a simple list of radio buttons.

I also see the cards had icons associated with them, but as far as I can tell this is only ever one icon. So it also makes sense to drop that.

**Comment by xianli123** [comment]
@jonkoops 

> as far as I can tell this is only ever one icon.

Yes, there is only one kind of mobile phone icon. When users set up OTP, they already know that they are using mobile. So there is no need to indicate the type of device in the OTP device options. 
<img width="729" alt="Screenshot 2023-06-16 at 17 08 09" src="https://github.com/keycloak/keycloak/assets/53931519/b0c29d08-95c3-416a-ab71-7410ef3f68f2">

**Comment by avelarix** [comment]
Hello all,It’s perfect!This is what I was thinking about!Thank youAvelarixSent from my iPhoneOn 16 Jun 2023, at 08:37, xianli123 ***@***.***> wrote:﻿
Hi @jonkoops @avelarix @rmartinc @R4BNL, Thanks for your suggestions. You may know that the new console is followed the design system named Patterfly 4. According to Patterfly4's guideline, I used a radio button instead of using the card view. The screenshot below shows the UI. What do you think?

**Comment by xianli123** [comment]
@avelarix Thanks for your reply. 

@jonkoops Could we implement this improvement and fix the issue [#21044](https://github.com/keycloak/keycloak/issues/21044)?  Here are some screens for you to look over below. Thanks!

<img width="856" alt="Screenshot 2023-06-19 at 10 44 46" src="https://github.com/keycloak/keycloak/assets/53931519/27f96ca9-5cb3-438d-9a77-5d124da16e0f">

**UI with measurements**
<img width="872" alt="Screenshot 2023-06-19 at 10 46 12" src="https://github.com/keycloak/keycloak/assets/53931519/6da2ee95-b6b6-4fab-800a-4663eee5a7db">

**Comment by R4BNL** [comment]
@xianli123 Thanks for your assistance with this, the new design looks very functional. For which version will this be possibly pushed?

**Comment by xianli123** [comment]
@avelarix My pleasure! 

> For which version will this be possibly pushed?

As for this question, what do you think? @jonkoops Thanks!

**Comment by jonkoops** [comment]
> For which version will this be possibly pushed? 

We don't give out estimates for that. It will be prioritized, and we will pick it up when we get to it. If this functionality lands it will be in the major version after that.

I expect this will be on the backlog for quite a while, as our team currently has larger priorities. The fastest possible way I see this landing is if we get a community contribution, so I will mark this with a 'help wanted' label.

**Comment by R4BNL** [comment]
@jonkoops Thanks for your prompt reply, that's no problem thanks for your help with this.

**Comment by jbman** [comment]
I like the improvement and would like to add a suggestion for a better headline, instead of "OTP devices". Usually the user knows the second factor as "Authenticator app", the abbreviation "OTP" isn't commonly known.
Proposal: **Authenticator app device** instead of **OTP devices**
